### PR TITLE
let user specify a number of processes for the local mini-runner

### PR DESCRIPTION
**Pull request by joezuntz**

### CODE REVIEW

**Review by empEvil — COMMENTED** [review]
2 things, I assume its not going to be an issue that in the 1 process case node would be (name, cores) before this change, now it will be (name_0, cores). Second, right now it makes n processes, and assigns them all the full count of cores available, is that intentional/ should it be doing that?

**Comment by joezuntz** [comment]
The name thing should be okay, as it's not actually used for anything, but vv good point on the cores, I will fix!

**Comment by joezuntz** [comment]
Okay, fixed!

File: ceci/sites/local.py [hunk 1]
```diff
@@ -79,11 +79,19 @@ def configure_for_parsl(self):
...
-        cores = psutil.cpu_count(logical=False)
-        cores = self.config.get("max_threads", cores)
+        # The default is to allow a single process
+        # with as many cores as possible, but allow the
+        # user to specify both max_processes and max_threads
+        # to customize
+        procs = self.config.get("max_processes", 1)
+        cores_available = psutil.cpu_count(logical=False)
+        threads_default = max(cores_available // procs, 1)
+        cores = self.config.get("max_threads", threads_default)
```
Comment (empEvil) [inline ceci/sites/local.py:89]: So you can still try to use `max_threads` to try and assign more cores to each process than available. I suppose that is ok?

Comment (joezuntz) [reply ceci/sites/local.py:89]: right - yes, for debugging or whatever the user wants to try.  Especially as for some systems you might want to use more threads than cores, depending on the architecture.

Comment (joezuntz) [reply ceci/sites/local.py:89]: If that make sense are you happy for me to merge?

Comment (empEvil) [reply ceci/sites/local.py:89]: yes, sounds good then, just wanted to make sure it was intended

Comment (joezuntz) [reply ceci/sites/local.py:89]: Thanks - will merge!

**Review by empEvil — APPROVED** [review]
It looks ready to merge then

**Pull request merged**